src/components/Product-card/ProductCard.jsx

```diff
@@ -1,10 +1,15 @@
 import "./productCard.css";
 import { FaRegHeart } from 'react-icons/fa';
 import { useProductsDetail } from "../../contexts/productsDetail-context";
+import { sortPriceFunc, sortRatingFunc, sortDiscountFunc, filterPriceFunc, filterSectionFunc, filterCategoryFunc, filterProductsFunc } from "../../utils/sortAndFilterFunc";
+import { useFilterData } from "../../contexts/filterData-context";
 
 const ProductCard = () => {
 
     const { products } = useProductsDetail();
+    const { state } = useFilterData();
```

Comment:
you could have used a better name instead of state.

Reply:
Just  a suggestion otherwise everything look good and structured

Reply:
yes, will work on it!! Thankyou for suggesting!